clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -792,12 +800,17 @@ private void timeoutCallsInFlight(TimeoutProcessor processor, Map<String, List<C
                 // only one we need to check the timeout for.
                 Call call = contexts.get(0);
                 if (processor.callHasExpired(call)) {
-                    log.debug("{}: Closing connection to {} to time out {}", clientId, nodeId, call);
-                    client.disconnect(nodeId);
-                    numTimedOut++;
-                    // We don't remove anything from the callsInFlight data structure.  Because the connection
-                    // has been closed, the calls should be returned by the next client#poll(),
-                    // and handled at that point.
+                    if (call.aborted) {
+                        log.warn("{}: aborted call {} is still in callsInFlight.", clientId, call);
```

Comment:
In the PR description, you said:

```text
It also adds the Call#aborted flag, which ensures that we only abort a connection once-- even if there is a similar bug in the future which causes old Call objects to linger.
```

It wasn't totally clear to me if you were suggesting that `abort` was useful in case we end up with a similar bug or if it was necessary for correctness under normal behaviour. It looks like the latter, is that correct?